### PR TITLE
feat(replays): Create new Other Sessions tab inside Replay Details page

### LINKED ISSUE #42817: Discuss: Link sessions from the same user together + navigate between them
**Solution Brainstorm**

- Chronological list of all sessions from X user. Could be another tab like "Console"
- Highlight which session is the current one the Sentry user is viewing

### PR DESCRIPTION
**Pull request by ryan953**
I switched from the normal 'session' column on the left to a simpler date/time column, because when the page loads for me, in my typical window size, the column is toooo narrow for the full url walker and the username is obviously redundant on each row.

The list loads more replays as you scroll.
The current replay is highlighted, there's no hover state or interactivity (yet?):

SCR-20230126-h5p

Fixes #42817

### CODE REVIEW

**Comment by Jesse-Box** [comment]
Minor thing @ryan953. Can we remove the calendar icon from the start time column (for both this tab and the replay index page?

Cheers

File: static/app/views/replays/detail/userReplays/index.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+];
+
+function ReplaysFromUser({replayId, user}: Props) {
+  const routes = useRoutes();
+  const organization = useOrganization();
+
+  const {eventView, loadRows} = useLoadReplaysFromUser({user});
+
+  const {isLoading, isRowLoaded, loadMoreRows, rows, rowState} =
+    useInfiniteLoader<ReplayListRecord>({
```
Comment (billyvg) [inline static/app/views/replays/detail/userReplays/index.tsx:49]: Note we don't use inf loading anywhere else in the application

**Comment by ryan953** [comment]
> Can we remove the calendar icon from the start time column (for both this tab and the replay index page?
> 
> Cheers

@Jesse-Box done: https://github.com/getsentry/sentry/pull/43252

File: static/app/views/replays/detail/userReplays/index.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/views/replays/detail/userReplays/index.tsx:49]: We kinda have the same ui pattern of scrolling down a long list though with the Replay network table, and console. Just that the data is pre-fetched. 

I had pagination buttons in here for a quick second, but it looked weird, the page is designed so the section extends flush to the bottom. we could design it another way, no rush to get this in.

File: static/app/views/replays/detail/userReplays/index.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+          </Body>
+        ) : null}
+      </CellMeasurer>
+    );
+  };
+
+  return (
+    <SessionsContainer>
+      <SessionsTable>
+        {isLoading ? (
```
[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/views/replays/detail/userReplays/index.tsx:104]: No it's not doing that. This is the loading state before the first set of data appears. I could look at this again though and put a 'loading...' message at the bottom of the table if someone is scrolling too fast, so they'll see rows and the message at the same time.

File: static/app/views/replays/replayTable/index.tsx [hunk 3]
```diff
@@ -12,15 +12,7 @@ import useOrganization from 'sentry/utils/useOrganization';
...
-import {
-  ActivityCell,
-  DurationCell,
-  ErrorCountCell,
-  ProjectCell,
-  SessionCell,
-  StartedAtCell,
-  TransactionCell,
-} from 'sentry/views/replays/replayTable/tableCell';
+import renderReplayCell from 'sentry/views/replays/replayTable/tableCell';
```
Comment (billyvg) [inline static/app/views/replays/replayTable/index.tsx:15]: The default export is named TableCell, can you make file == default export name == default import name

**Review by billyvg — COMMENTED** [review]
Can we center the cells (and possibly fix the column width fluctuations) before merging?

Oh and we should have a button to open up the other session.


The other feedback we can talk about and/or ticket for future work.

**Comment by Jesse-Box** [comment]
@ryan953 here's some more feedback

As a user looking at this component, I'd say they are more interested in metadata of the session rather than user details or the number of pages they visited. With that in mind, I would, reduce the width of the user column, so the other columns can breathe. The goal should be to remove any horizontal scroll, so let's make sure that the user column only takes up the available space, if it's too narrow, then the overflow doesn't wrap, rather is covered by ellipses.

If I remember my grid CSS, it would be something like
`grid-column-templates: 1fr repeat(max-content, 4)`

File: static/app/views/replays/detail/userReplays/index.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+import {ReplayColumns} from 'sentry/views/replays/replayTable/types';
+import type {ReplayListRecord, ReplayRecord} from 'sentry/views/replays/types';
+
+type Props = {
+  replayId: undefined | string;
+  user: undefined | ReplayRecord['user'];
+};
+
+const COLUMNS = [
+  ReplayColumns.user,
```
Comment (ryan953) [inline static/app/views/replays/detail/userReplays/index.tsx:35]: This column isn't the best; it's really repetitive to see the username over and over again. 

Maybe if it was just an icon, or if the icon was at the top and then this was something else... but right now it's our link into the replay that the row represents.

File: static/app/views/replays/replayTable/index.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/views/replays/replayTable/index.tsx:15]: `tableCell.tsx` and `TableCell` all the things 👍

**Pull request closed without merging**